File: leet_BST_0938_rangesum___Recur.go
func rangeSumBST(node *TreeNode, lo int, hi int) int {
	if node == nil {
		return 0
	}
	val := node.Val
	if val < lo {
		return rangeSumBST(node.Right, lo, hi)
	}
	if val > hi {
		return rangeSumBST(node.Left, lo, hi)
	}
	return val + rangeSumBST(node.Right, lo, hi) + rangeSumBST(node.Left, lo, hi)
}

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

File: leet_BST_0938_rangesum___Recur.py
class Solution:
    def rangeSumBST(self, node: Optional[TreeNode], lo: int, hi: int) -> int:
        if not node or lo > hi:
            return 0
        if node.val < lo:
            return self.rangeSumBST(node.right, lo, hi)
        if node.val > hi:
            return self.rangeSumBST(node.left, lo, hi)
        r = node.val + self.rangeSumBST(node.right, lo, hi) + self.rangeSumBST(node.left, lo, hi)
        return r

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
